### PR TITLE
AP_HAL_ChibiOS: Update luminousbee5 hwdefs, Add new bootloaders

### PR DESCRIPTION
**Pull request by DavideLentini**
Hi everyone @rmackay9 @Hwurzburg , I wanted to create another PR to share the new hwdefs for luminousbee5 . 
Following Randy's advice , I squashed the commits and called AP_HAL_ChibiOS and created new bootloaders for luminousbee5 (bin , hex) . This way everything is neater :)

### CODE REVIEW

**Comment by rmackay9** [comment]
Thanks!  I'm not actually qualified to review this but I've added it to the agenda for next week's dev call.

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:214]: this will break things for any user using an older board with a BMI088. Is that really what you want to do? 
I think we should make it work with both older and newer boards

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef-bl.dat
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef-bl.dat:32]: this order doesn't matter

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:104]: why change to FLOAT? This could lead to bad detection if a receiver is not connected

**Review by tridge — CHANGES_REQUESTED** [review]
there are enough changes that I think you should consider making this a new hwdef, maybe liminousbee5A ? or 5.1 ?

**Comment by tridge** [comment]
we could (with a bit of extra effort) make one fw that works with both versions, including with the batt param changes, or we could do a separate fw

**Comment by DavideLentini** [comment]
@tridge 
Thank you very much ,for your help and analysis of our hwdef . Before moving on to the analysis of the problems you found , I would like to discuss our idea to keep the luminousbee5 hwdef folder :

The first version of luminousbee was 5.1 .
The current version is 5.32 .
The production version instead will be 5.4 .

Since these versions all have the prefix "5." , our idea was to keep the folder "luminousbee5" for all hwdef versions of these boards and the name "luminousbee5" for bootloaders, until the version reaches the prefix "6.".

The boards that will most likely go out to eventual customers , will be either the 5.32 or the 5.4 . The difference between these two boards (visible in the hwdef) is the IMU which has been changed from 42605 to 42688. However, we managed to integrate the 2 IMUs in the same hwdef , so this hwdef can be used to compile a firmware for both 5.32 and 5.4 . 

So we would like to keep the same folder "luminousbee5" , for updates of luminousbee 5.X versions . If this creates problems or conflicts with ardupilot's hwdefs system , we can discuss it to find a solution :)

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:104]: Mm . I don't know exactly why this was changed from PULLDOWN to FLOAT . 
Also because you had yourself changed it in 5.1 with this commit :
https://github.com/ArduPilot/ardupilot/blob/47da7f5c9b5913ac12f95713c7ff08e7621c330c/libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
I will check with my hardware colleagues , but if you tell us that it is better to keep it at PULLDOWN , we will change it right away .

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef-bl.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef-bl.dat:32]: We had changed this for an experimental feature .We would like the bootloader to give priority to USART2 , corresponding to ESP8266 before USB .
So far this has not affected other bootloader operations and kept our feature working .

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:104]: @tridge 
Hello , then you are right ! It must have been an oversight going between versions . I will change it

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:214]: In this case as I have already written , in my last comment . The new boards will not use this IMU . So keeping it commented out should not create future problems .

**Comment by Hwurzburg** [comment]
will need a squash, and then a Tools/gittools/git-sub-systems-split  use , and a forced push over the existing branch to get the correct split of commits

**Comment by DavideLentini** [comment]
> will need a squash, and then a Tools/gittools/git-subsystems-split use , and a forced push over the existing branch to get the correct split of commits
Understood . Sorry but it is my first PR in this repo . What is the function of the `Tools/gittools/git-subsystems-split tool` ?

**Comment by Hwurzburg** [comment]
it will create a separate commit for each library...in this case: one for Tools and one for AP_HAL_ChibiOS...the standard is to have one commit per library section or vehicle sub-directory per PR

**Comment by Hwurzburg** [comment]
@DavideLentini let me know when you are finished making changes and I will put a DevCall tag on it for dev team review for merging...

**Comment by DavideLentini** [comment]
> @DavideLentini let me know when you are finished making changes and I will put a DevCall tag on it for dev team review for merging...

Thanks , yes the 2 hwdefs are ready, and the bootloaders too

**Comment by DavideLentini** [comment]
> @DavideLentini let me know when you are finished making changes and I will put a DevCall tag on it for dev team review for merging...

Yes , probably we'll make a final review before the end of this week

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:2]: Can we have comments in English please?

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:5]: This now should be
```
MCU_CLOCKRATE_MHZ 480
```

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:190]: You usually want the lower speed to be safe. So maybe this should be something like 2*Mhz 16*Mhz

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:2]: Sure !

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:5]: I will report this to my colleague , thanks

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat:190]: I will report this to my colleague , thanks

Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat]: Resolved now

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat]: Resolved Now

File: libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat
Comment (DavideLentini) [reply libraries/AP_HAL_ChibiOS/hwdef/luminousbee5/hwdef.dat]: Resolved now

**Review by andyp1per — APPROVED** [review]
LGTM

**Comment by DavideLentini** [comment]
Hi @tridge , this PR has been open for a while , I am waiting for your review please . Thank you very much for your patience

**Review by tridge — APPROVED** [review]

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
sorry,I removed the mergeonCIpass until these questions are addressed:
1. if this flight controller is to be used by anyone else than the designer or his company,and is to be listed in  the wiki then an adequate README.md must be included....if for private use onl then it not needed.
2. you pushed again ... after the approval.....rebasing on master is fine but you did not squash

**Comment by DavideLentini** [comment]
@Hwurzburg 
Sorry , small inattention . I squashed and pushed . 
To answer your first question , no there is no need, it's private . :)

**Review by Hwurzburg — APPROVED** [review]

**Review by Hwurzburg — APPROVED** [review]

**Pull request merged**
